commit f738eb1b63edf664da1b4ac76895d988749b2f07
Author: Ingo Molnar <mingo@elte.hu>
Date:   Tue Aug 18 11:32:24 2009 +0200

    perf_counter: Fix the PARISC build
    
    PARISC does not build:
    
    /home/mingo/tip/kernel/perf_counter.c: In function 'perf_counter_index':
    /home/mingo/tip/kernel/perf_counter.c:2016: error: 'PERF_COUNTER_INDEX_OFFSET' undeclared (first use in this function)
    /home/mingo/tip/kernel/perf_counter.c:2016: error: (Each undeclared identifier is reported only once
    /home/mingo/tip/kernel/perf_counter.c:2016: error: for each function it appears in.)
    
    As PERF_COUNTER_INDEX_OFFSET is not defined.
    
    Now, we could define it in the architecture - but lets also provide
    a core default of 0 (which happens to be what all but one
    architecture uses at the moment).
    
    Architectures that need a different index offset should set this
    value in their asm/perf_counter.h files.
    
    Cc: Kyle McMartin <kyle@mcmartin.ca>
    Cc: Helge Deller <deller@gmx.de>
    Cc: linux-parisc@vger.kernel.org
    Cc: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Cc: Mike Galbraith <efault@gmx.de>
    Cc: Paul Mackerras <paulus@samba.org>
    Cc: Arnaldo Carvalho de Melo <acme@redhat.com>
    Cc: Frederic Weisbecker <fweisbec@gmail.com>
    LKML-Reference: <new-submission>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/kernel/perf_counter.c b/kernel/perf_counter.c
index b8fe739..36f65e2 100644
--- a/kernel/perf_counter.c
+++ b/kernel/perf_counter.c
@@ -2019,6 +2019,10 @@ int perf_counter_task_disable(void)
 	return 0;
 }
 
+#ifndef PERF_COUNTER_INDEX_OFFSET
+# define PERF_COUNTER_INDEX_OFFSET 0
+#endif
+
 static int perf_counter_index(struct perf_counter *counter)
 {
 	if (counter->state != PERF_COUNTER_STATE_ACTIVE)
